worlds/saving_princess/Items.py

```diff
@@ -0,0 +1,92 @@
+from typing import Optional, Dict
+
+from BaseClasses import Item, ItemClassification as ItemClass
+
+from .Constants import *
+
+
+class SavingPrincessItem(Item):
+    game: str = GAME_NAME
+
+
+class ItemData:
+    item_class: ItemClass
+    code: Optional[int]
+    count: int  # Number of copies for the item that will be made of class item_class
+    count_extra: int  # Number of extra copies for the item that will be made as useful
+
+    def __init__(self, item_class: ItemClass, code: Optional[int] = None, count: int = 1, count_extra: int = 0):
+        self.item_class = item_class
+
+        self.code = code
+        if code is not None:
+            self.code += BASE_ID
+
+        if self.item_class == ItemClass.filler or code is None:
```

Comment:
I don't believe this `if` block ever does anything because all your filler and victory items explicitly set their `count` to 0 and the default for `count_extra` is already 0

Reply:
I agree, it is redundant and I should decide on one or the other.

I personally think that keeping the `if` block (and adding the check for `ItemClass.trap` that it is missing) while removing the redundant arguments in the item data dicts would look cleaner.